utility/src/main/java/org/oppia/android/util/math/RealExtensions.kt

```diff
@@ -0,0 +1,84 @@
+package org.oppia.android.util.math
+
+import org.oppia.android.app.model.Real
+import org.oppia.android.app.model.Real.RealTypeCase.INTEGER
+import org.oppia.android.app.model.Real.RealTypeCase.IRRATIONAL
+import org.oppia.android.app.model.Real.RealTypeCase.RATIONAL
+import org.oppia.android.app.model.Real.RealTypeCase.REALTYPE_NOT_SET
+
+/**
+ * Returns whether this [Real] is explicitly a rational type (i.e. a fraction).
+ *
+ * This returns false if the real is an integer despite that being mathematically rational.
+ */
+fun Real.isRational(): Boolean = realTypeCase == RATIONAL
+
+/** Returns whether this [Real] is negative. */
+fun Real.isNegative(): Boolean = when (realTypeCase) {
+  RATIONAL -> rational.isNegative
+  IRRATIONAL -> irrational < 0
+  INTEGER -> integer < 0
+  REALTYPE_NOT_SET, null -> throw IllegalStateException("Invalid real: $this.")
+}
+
+/**
+ * Returns a [Double] representation of this [Real] that is approximately the same value (per
+ * [isApproximatelyEqualTo]).
+ */
+fun Real.toDouble(): Double {
+  return when (realTypeCase) {
+    RATIONAL -> rational.toDouble()
+    INTEGER -> integer.toDouble()
+    IRRATIONAL -> irrational
+    REALTYPE_NOT_SET, null -> throw IllegalStateException("Invalid real: $this.")
+  }
+}
+
+/**
+ * Returns a human-readable, plaintext representation of this [Real].
+ *
+ * Note that the returned value is guaranteed to be a self-contained numeric expression representing
+ * the real (which means proper fractions are converted to improper answer strings since fractions
+ * like '1 1/2' can't be written as a numeric expression without converting them to an improper
+ * form: '3/2').
+ */
+fun Real.toPlainText(): String = when (realTypeCase) {
+  // Note that the rational part is first converted to an improper fraction since mixed fractions
+  // can't be expressed as a single coefficient in typical polynomial syntax).
+  RATIONAL -> rational.toImproperForm().toAnswerString()
+  IRRATIONAL -> irrational.toPlainString()
+  INTEGER -> integer.toString()
+  REALTYPE_NOT_SET, null -> ""
```

Comment:
Can we use some const here, making it more readable and understandable that why we are passing an empty string here?

Reply:
I added a KDoc and inline comment to contextualize this--is it clearer?